reVX/handlers/geotiff.py

```diff
@@ -64,14 +64,16 @@ def __getitem__(self, keys):
             Slicing args similar to a numpy array slice. See examples above.
         """
         ds, ds_slice = parse_keys(keys)
-
-        if ds == 'meta':
-            out = self._get_meta(*ds_slice)
-        elif ds.lower().startswith('lat'):
-            out = self._get_lat_lon(*ds_slice)[0]
-        elif ds.lower().startswith('lon'):
-            out = self._get_lat_lon(*ds_slice)[1]
-        else:
+        out = None
+        if isinstance(ds, str):
+            if ds == 'meta':
+                out = self._get_meta(*ds_slice)
+            elif ds.lower().startswith('lat'):
+                out = self._get_lat_lon(*ds_slice)[0]
+            elif ds.lower().startswith('lon'):
+                out = self._get_lat_lon(*ds_slice)[1]
+
+        if out is None:
```

Comment:
How about just `else:` for line 76? Should this throw an error before returning `None`?

Reply:
So the issue you ran into before is that the if/elif/elif statement ASSUMES `ds` is a string, but the bands are normally numerical. The most verbose way of doing this would be:
```
        ds, ds_slice = parse_keys(keys)

        if isinstance(ds, str):
            if ds == 'meta':
                out = self._get_meta(*ds_slice)
            elif ds.lower().startswith('lat'):
                out = self._get_lat_lon(*ds_slice)[0]
            elif ds.lower().startswith('lon'):
                out = self._get_lat_lon(*ds_slice)[1]
           else:
                out = self._get_data(ds, *ds_slice)
       else:
            out = self._get_data(ds, *ds_slice)

        return out
```

But that seems silly redudant, instead I just check for meta, lat, or lon, otherwise just pull the given ds value

Reply:
 👍